cagnotte_base/models/cagnotte.py

```diff
@@ -0,0 +1,202 @@
+            on cagnotte
+        """
+        for cagnotte in self:
+            if cagnotte.product_id:
+                product_account_id = cagnotte.product_id.\
+                    property_account_income_id.id
+                if not product_account_id:
+                    product_account_id = cagnotte.product_id.categ_id.\
+                        property_account_income_categ_id.id
+                if not product_account_id or \
+                        product_account_id != cagnotte.account_id.id:
+                    return False
+            journal_debit_account_id = cagnotte.journal_id.\
+                default_debit_account_id.id
+            if journal_debit_account_id != cagnotte.account_id.id:
+                return False
+            journal_credit_account_id = cagnotte.journal_id.\
+                default_credit_account_id.id
+            if journal_credit_account_id != cagnotte.account_id.id:
+                return False
+        return True
+
+    _constraints = [
+        (_check_account,
+         'Accounts not corresponding between product, journal and cagnotte',
+         ['product_id', 'account_id', 'journal_id']),
+    ]
+
+    _sql_constraints = [(
+        'product_cagnotte_uniq',
+        'unique(product_id, company_id)',
+        'A cagnotte type with the product already exist'
+    ), (
+        'account_cagnotte_uniq',
+        'unique(account_id)',
+        'A cagnotte type with this account already exist'
+    )]
+
+
+class AccountCagnotte(models.Model):
+    _name = 'account.cagnotte'
+
+    name = fields.Char(
+        readonly=True,
+        copy=False)
+    cagnotte_type_id = fields.Many2one(
+        'cagnotte.type',
+        'Cagnotte Type',
+        required=True,
+        ondelete='restrict')
+    solde_cagnotte = fields.Float(
+        compute='_compute_solde_cagnotte',
+        store=True)
+    active = fields.Boolean(
+        default=True)
+    account_move_line_ids = fields.One2many(
+        comodel_name="account.move.line",
+        inverse_name="account_cagnotte_id",
+        string="Journal Items")
+    create_date = fields.Date(
+        default=fields.Date.today)
+
+    @api.multi
+    def _get_name(self):
+        self.ensure_one()
+        return '%s - %s' % (self.cagnotte_type_id.name, self.name)
+
+    @api.multi
+    def name_get(self):
+        """Add the type of cagnotte in the name"""
+        res = []
+        for cagnotte in self:
+            name = cagnotte._get_name()
+            res.append((cagnotte.id, name))
+        return res
+
+    @api.multi
+    @api.depends('account_move_line_ids.debit',
+                 'account_move_line_ids.credit')
+    def _compute_solde_cagnotte(self):
+        for cagnotte in self:
+            solde_cagnotte = 0
+            for move_line in cagnotte.account_move_line_ids:
+                solde_cagnotte += move_line.credit - move_line.debit
+            cagnotte.solde_cagnotte = solde_cagnotte
```

Comment:
@Cedric-Pigeon Performances would be better with a read_group.

```python
        @api.multi
        @api.depends('account_move_line_ids.debit',
                                 'account_move_line_ids.credit')
        def _compute_solde_cagnotte(self):
        aml_model = self.env['account.move.line']
        domain = [
            ('account_cagnotte_id', 'in', self.ids)]

        res = aml_model.read_group(
            domain=domain,
            fields=['account_cagnotte_id', 'credit', 'debit'],
            groupby=['account_cagnotte_id']
        )

        aml_dict = {}
        for dic in res:
            cagnotte_id = dic['account_cagnotte_id'][0]
            solde_cagnotte = dic['credit'] - dic['debit']
            aml_dict[cagnotte_id] = solde_cagnotte

        for cagnotte in self:
            cagnotte.solde_cagnotte = aml_dict.get(cagnotte.id, 0.0)
```

But `solde_cagnotte` is stored so no big deal I guess...